pkg/otlp/model/translator/metrics_translator.go

```diff
@@ -33,6 +34,14 @@ import (
 
 const metricName string = "metric name"
 
+var _ source.Provider = (*noSourceProvider)(nil)
+
+type noSourceProvider struct{}
+
+func (*noSourceProvider) Source(context.Context) (source.Source, error) {
+	return source.Source{Kind: source.HostnameKind, Identifier: ""}, nil
```

Comment:
Do we need a `source.KindUnknown` type specifically for this?

Reply:
I think the right approach is to enforce passing the sourcer as a required argument in `translator.New`, to avoid this whole thing. I will do this together with the other changes on a future PR, since it affects public API